srcs/expansion/command_substitution.c

```diff
@@ -91,13 +90,14 @@ static t_strlist			*split_subsitutions(char const *word)
 	return (result);
 }
 
-char						*command_substition(char const *word)
+t_strlist		*split_subsitutions(char const *word)
 {
-	t_strlist	*strlist;
-	char		*result;
+	char const	*passv_str_start;
+	t_strlist	*result;
+	char		quoted;
 
-	strlist = split_subsitutions(word);
-	result = strlist_to_str(strlist);
-	strlist_delete(&strlist);
-	return (result);
+	quoted = 0;
+	passv_str_start = NULL;
+	result = NULL;
+	return (split_subsitutions_run(word, passv_str_start, result, quoted));
```

Comment:
tu passe une copie de la variable, ça va rien casser que ce soit une copie passé en argument ou une variable déclarée dans la fonction c'est la même chose. 
http://denniskubes.com/2012/08/20/is-c-pass-by-value-or-reference/